vertx-core/src/main/java/io/vertx/core/spi/cluster/ClusterManager.java

```diff
@@ -52,14 +52,10 @@ default void init(VertxBootstrap builder) {
 
   /**
    * Invoked before this cluster node tries to join the cluster.
-   * <p>
-   * Implementations must signal the provided {@code nodeSelector} when messaging handler registrations are added or removed
-   * by sending a {@link RegistrationUpdateEvent} with {@link NodeSelector#registrationsUpdated(RegistrationUpdateEvent)}.
    *
    * @param vertx        the Vert.x instance
-   * @param nodeSelector the {@link NodeSelector} that must receive {@link RegistrationUpdateEvent}.
    */
-  void init(Vertx vertx, NodeSelector nodeSelector);
+  void init(Vertx vertx);
```

Comment:
I don't understand why the `RegistrationListener` must be set via a separate method. Why can't it replace the `NodeSelector` as a parameter?

Reply:
we have `nodeListener(NodeListener listener)`, following the same pattern with `RegistrationListener`